### ISSUE TITLE
When switching offline maps, previous map is still on screen at current zoom level

### ISSUE THREAD

**Issue by cachly** [cachly/bug-tracker#1176]
**[Original report](https://bitbucket.org/zedsaidstudio/cachly/issues/1176) by Zed Said Studio LLC (Bitbucket: [zedsaid](https://bitbucket.org/zedsaid), GitHub: [zedsaid](https://github.com/zedsaid)).**

----------------------------------------

Possibly started in iOS 10?

**Comment by cachly** [comment]
**Original comment by Zed Said Studio LLC (Bitbucket: [zedsaid](https://bitbucket.org/zedsaid), GitHub: [zedsaid](https://github.com/zedsaid)).**

----------------------------------------

Cachly 2.0 (21) on iPhone 6 running ios 10.0.2. I live on the border of NJ/NY and cache in both states, often on the same day. I have the offline maps loaded for both states. 

Behavior 1:
I can reproduce Cachly beta force closing with the following:
- Cachly has downloaded offline maps for NJ & NY.  These have been downloaded on my device throughout most of the beta. 
- I display a bookmark list which contains 7 caches crossing the border of NY & NJ
- On the main screen I click to switch the underlying map. Offline NJ is displayed and I select offline NY which causes Cachly to force close.

Behavior 2:
If I launch Cachly beta with the initial Live map displayed and the NJ offline map displayed, switching the underlying map to a different offline map such as NY *sometimes* works but more often then not the displayed map remains unchanged. If I click the icon to switch maps again, it shows the other map is selected but the display is still showing the other (NY selected - NJ displayed).
If I have an offline map displayed and select an online map, it always switches to the online map, but when I switch back to a different offline map from what was previously selected, the map still shows the other offline map...
Eg:
1) Offline NY was selected and that map is displayed
2) Online Google map selected and display updated
3) Offline NJ selected and map changes but map displayed offline NY

Behavior 3:
1 & 2 above are what I saw but I also noticed that if I moved the map display after switching the offline maps, *new* map tiles would load for the selected state and *new* map tiles wouldn't load for the other state. But Cachly isn't loading the map segments for any of the tiles that are displayed on the screen when the offline map is changed. If I zoom out, it loads the new map segment for the selected state but if I zoom back in again, the map segment is again blank.  

Hope this helps identify the general problem area.
